fedimint-sled/src/lib.rs

```diff
@@ -117,14 +119,20 @@ impl IDatabase for SledDb {
     }
 
     fn begin_transaction(&self) -> DatabaseTransaction {
-        SledTransaction {
+        let mut tx: DatabaseTransaction = SledTransaction {
             operations: Vec::new(),
             db: self,
+            num_pending_operations: 0,
+            num_savepoint_operations: 0,
         }
-        .into()
+        .into();
+        tx.set_tx_savepoint();
+        tx
     }
 }
 
+// Sled database transaction should only be used for test code and never for production
+// as it doesn't properly implement MVCC
```

Comment:
Not totally related, but I think there's an unforced error in the sled tx impl. The for loop should terminate early. Currently an insert after delete would lead to the insert being swallowed imo.

Reply:
Isn't that the behavior that we want though? If you have a database transaction that hasn't committed yet, then you do an insert, then you do a delete, then you do a read of that same key, the database should tell you that the key doesn't exist right? Let me know if I'm missing something.